### ISSUE TITLE
Remove `data_loss_ratio` and `data_loss_ratio_blinks` from parsed metadata

### ISSUE THREAD

**Issue by cbueth** [aeye-lab/pymovements#1584]
Remove `pre_processed_metadata['data_loss_ratio']` and `pre_processed_metadata['data_loss_ratio_blinks']` fields from `pre_processed_metadata`.

https://github.com/pymovements/pymovements/blob/95a30e2cdfa21bf1e0b53fe955390137da70e63c/src/pymovements/gaze/_utils/_parsing_eyelink.py#L891-L894

- [ ] Remove the two fields from metadata dict
- [x] Test equality using measure_event_ratio and `pymovements.measure.samples.measures.data_loss`.
- [ ] see if to be removed parsing tests have interesting test cases for measure_event_ratio

**Comment by cbueth** [comment]
**TL;DR** Yes, we can remove and are on par. This issue proposes checking for equality, so we can reassure that applications using the private metadata can safely migrate to the existing, user-facing replacement functions.

# Data Loss Review: Metadata Dict vs. Measure

## Context

This issue proposes removing `pre_processed_metadata['data_loss_ratio']` and `['data_loss_ratio_blinks']` as part of handling the metadata dict issues
- #893,
- #1082, and
- #1398.

The replacements already exist:

| Metadata field                    | Replacement after removal                         | Source                    |
| --------------------------------- | ------------------------------------------------- | ------------------------- |
| `data_loss_ratio`                 | `data_loss(column, sampling_rate=..., unit='ratio')` on per-trial groups | `pymovements.measure`     |
| `data_loss_ratio_blinks`          | `measure_events_ratio('blink', sampling_rate=...)` on per-trial groups   | `Gaze` method (#1420)  |

---

## How Both Approaches Calculate Data Loss

### 1. Metadata `data_loss_ratio` and `data_loss_ratio_blinks` 

Both EyeLink parser only:

```
n_expected  = Σ round(block_duration_ms * sr / 1000)    [from START/STOP recording messages]
n_valid     = count of rows where ALL gaze coords are non-NaN AND not in blink
ratio       = (n_expected - n_valid) / n_expected
```

```
blink_intervals = merge overlappping blink events
n_blink_samples = Σ round((offset - onset) / sample_length) + 1   [sample_length = 1000/sr]
ratio           = n_blink_samples / n_expected
```

### 2. `data_loss()` measure (format-independent)

```
expected    = floor((end - start) * sr / 1000) + 1          [per group/trial]
time_loss   = max(expected - observed_row_count, 0)
invalid_loss = Σ null/NaN/inf values in data column
ratio       = (time_loss + invalid_loss) / expected
```

### 3. `measure_events_ratio('blink')` (Gaze method, delegates to `events2timeratio`)

The `+dt` inclusive-endpoint convention is controlled by the `sampling_rate` argument:

```
# With sampling_rate -> inclusive (+dt):
dt         = 1000 / sampling_rate
numerator  = Σ (offset - onset + dt)
denominator = max(time) - min(time) + dt

# Without sampling_rate -> non-inclusive:
numerator  = Σ (offset - onset)
denominator = max(time) - min(time)
```

Without `sampling_rate`, the method falls back to estimating `dt` from mode of time diffs, which does **not** trigger the `+dt` convention. The formula is then `Σ(offset - onset) / span`.

---

## Empirical Equivalence on Real Data

I tested this with five files, about 216 trials each, at 1000Hz. They are identical.

### Total data_loss_ratio

| File            | Metadata    | `data_loss()` per-block | \(\Delta\)    |
| ----------- | :---------: | :---------------------: | :-----------: |
| [redacted] | 0.005249    | 0.005249                | 0.000000  |
| [redacted] | 0.004996    | 0.004996                | 0.000000  |
| [redacted] | 0.016995    | 0.016995                | 0.000000  |
| [redacted] | 0.060254    | 0.060254                | 0.000000  |
| [redacted] | 0.001877    | 0.001877                | 0.000000  |

### Blink data_loss_ratio

| File            | Metadata    | `measure_events_ratio` | \(\Delta\)    |
| ----------- | :---------: | :--------------------: | :-----------: |
| [redacted] | 0.005243    | 0.005243               | 0.000000  |
| [redacted] | 0.004978    | 0.004978               | 0.000000  |
| [redacted] | 0.016968    | 0.016969               | 0.000001  |
| [redacted] | 0.060196    | 0.060201               | 0.000005  |
| [redacted] | 0.001848    | 0.001848               | 0.000000  |

There is a negligible difference, but this can be imrpoved:

---

## Size and Relevance of the Blink Difference

Both formulas compute the same underlying quantity, the fraction of time covered by blink events, but use different arithmetic. They become equivalent when each trial spans exactly one recording block:

```
metadata:  Σ round((offset - onset)/dt + 1)  /  Σ n_expected       [sample count / sample count]
measure:  Σ (offset - onset + dt)           /  Σ (span + dt)       [time / time]
```

Within a single trial, `span = tmax - tmin = recording_block_duration` and `n_expected = round(span x sr/1000) ≈ span/dt`. Substituting:

```
metadata:  (round(D/dt) + 1) · dt / (round(span/dt) · dt)  ≈  (D + dt) / (span + dt)  :measure
```

The approximation `round(x) ≈ x` and the `+dt` given by `sampling_rate=` make both sides converge. The formulas are algebraically identical for integer-valued spans and durations.

Passing `sampling_rate` triggers the `+dt` inclusive-endpoint convention needed to match the metadata's `round() + 1` sample counting. Without it, the measure uses non-inclusive endpoints and differs systematically:

| File            | no `sampling_rate` | With `sampling_rate` | Metadata   | \(\Delta\) (with +sr) |
| ---------- | :----------------: | :------------------: | :--------: | :-------------------: |
| [redacted] |      0.005206      |       0.005242       |  0.005243  |       0.000001        |
| [redacted] |      0.004920      |       0.004977       |  0.004978  |       0.000001        |
| [redacted] |      0.016846      |       0.016967       |  0.016968  |       0.000001        |
| [redacted] |      0.059661      |       0.060195       |  0.060196  |       0.000000        |
| [redacted] |      0.001836      |       0.001847       |  0.001848  |       0.000000        |

The residual deltas come from the test scripts approximating trial boundaries with a 1s time-gap heuristic instead of the actual recording blocks. With real trial columns:

```python
Gaze.samples.group_by(trial_cols).agg(
    Gaze.measure_events_ratio('blink', sampling_rate=sr)
)
```

where each trial spans precisely one START->STOP recording block, the formulas are algebraically equivalent and the results are identical.


---

## Edge Cases

While trying my test code on `eyelink_binocular_example.asc`, I got `data_loss_ratio = -0.1245` in metadata because:
```
n_expected = 241   (from 241ms recording block)
n_valid    = 271   (more valid samples than expected!)
```
`(241 - 271) / 241 = -0.1245`

The `num_expected_samples` formula (`round(block_duration * sr / 1000)`) undercounts when recording duration and block boundaries don't align. The `data_loss()` measure returns a plausible 21.74% on the same file, which is the correct value (NaN loss from the right-eye column).

Furthermore, for `eyelink_monocular_no_dummy_example.asc` having no START/STOP recording blocks, metadata returns `None` for both fields. `data_loss()` returns 23.23% valid result. That is because the measure works without recording config.

---

## How to migrate

`pre_processed_metadata['data_loss_ratio']` -> `Gaze.samples.group_by(trial_cols).agg(data_loss(column, sampling_rate=sr, unit='ratio'))`
`pre_processed_metadata['data_loss_ratio_blinks']` -> `Gaze.samples.group_by(trial_cols).agg(Gaze.measure_events_ratio('blink', sampling_rate=sr))`

### For this to work

1. Trial/block boundaries must be available, so the measure can use `min(time)/max(time)` per group, so grouping by trial/block is essential.

2. `sampling_rate` must be available. stored in `pre_processed_metadata['sampling_rate']`, this should be accessible via `Gaze.experiment.sampling_rate` or a similar public attribute.

3. `_calculate_data_loss_ratio()` in `src/pymovements/gaze/_utils/_parsing.py` can be deleted once removed from `parse_eyelink`.

### Pro/Con

The measure implemented in #1389 is independent of the ASC/EDF format, supports any Gaze DataFrame, independent of START/STOP messages, is groupable per trial and not dependent per file and supports several output units. The only advantage the metadata result has is the blink overlap merge, which measure_events_ratio doesn't deduplicate by default.

### Summary

Both metadata fields can be safely removed. The measure-based replacements produce functionally identical results when used per-trial with a known sampling rate. The `measure_events_ratio` approach is slightly cleaner (time-based rather than sample-count arithmetic) and avoids the negative-ratio bug. A PR removing the metadata fields should also migrate from the parsing tests to measure tests.
